pkg/daemon/agent/flexvolume/server.go

```diff
@@ -106,30 +109,47 @@ func (s *FlexvolumeServer) Start() error {
 }
 
 // Stop the unix domain socket server and deletes the socket file
-func (s *FlexvolumeServer) Stop() {
-	if s.listener != nil {
-		logger.Info("Stopping unix socket rpc server.")
-		if err := s.listener.Close(); err != nil {
+func (s *FlexvolumeServer) StopAll() {
+	logger.Infof("Stopping %d unix socket rpc server(s).", len(s.listeners))
+	for unixSocketFile, listener := range s.listeners {
+		if err := listener.Close(); err != nil {
 			logger.Errorf("Failed to stop unix socket rpc server: %+v", err)
 		}
+
+		// closing the listener should remove the unix socket file. But lets try it remove it just in case.
+		if _, err := os.Stat(unixSocketFile); !os.IsNotExist(err) {
+			logger.Infof("Deleting unix domain socket file %s.", unixSocketFile)
+			os.Remove(unixSocketFile)
+		}
 	}
-	// closing the listener should remove the unix socket file. But lets try it remove it just in case.
-	unixSocketFile := path.Join(flexVolumeDriverDir, path.Join(UnixSocketName)) // /flextmnt/rook.io~rook/.rook.sock
-	if _, err := os.Stat(unixSocketFile); !os.IsNotExist(err) {
-		logger.Info("Deleting unix domain socket file.")
-		os.Remove(unixSocketFile)
-	}
+	s.listeners = make(map[string]net.Listener)
+}
 
+func RookDriverName(context *clusterd.Context) (string, error) {
+	kubeVersion, err := k8sutil.GetK8SVersion(context.Clientset)
+	if err != nil {
+		return "", fmt.Errorf("Error getting server version: %v", err)
+	}
+	if kubeVersion.AtLeast(version.MustParseSemantic(serverVersionV180)) {
+		// the driver name needs to be the same as the namespace so that we can support multiple namespaces
+		// without the drivers conflicting with each other
+		return os.Getenv(k8sutil.PodNamespaceEnvVar), nil
+	}
+	// fall back to the legacy driver name where multiple system namespaces are not supported
+	return LegacyFlexDriverName, nil
```

Comment:
should we always fall back to using the legacy driver name, instead of this function potentially returning an error?

Reply:
i pondered that question as well. We need a driver name that is predictable so that everything will be hooked up correctly both in the agent and in the operator. If we fall back to the `rook` name, we can then run into the conflict between rook-system namespaces, which we are trying to get away from. Therefore, it seemed better to fail early and rely on K8s retry mechanisms